x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/rerank/TextSimilarityRankBuilder.java

```diff
@@ -0,0 +1,125 @@
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ml.rerank;
+
+import org.apache.lucene.search.Explanation;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.search.rank.RankBuilder;
+import org.elasticsearch.search.rank.RankDoc;
+import org.elasticsearch.search.rank.context.QueryPhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.context.QueryPhaseRankShardContext;
+import org.elasticsearch.search.rank.context.RankFeaturePhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.context.RankFeaturePhaseRankShardContext;
+import org.elasticsearch.search.rank.rerank.RerankingQueryPhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.rerank.RerankingQueryPhaseRankShardContext;
+import org.elasticsearch.search.rank.rerank.RerankingRankFeaturePhaseRankShardContext;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+
+public class TextSimilarityRankBuilder extends RankBuilder {
+
+    private final String inferenceId;
+    private final String inferenceText;
+    private final String field;
+    private final float minScore;
+
+    public TextSimilarityRankBuilder(String field, String inferenceId, String inferenceText, int rankWindowSize, float minScore) {
+        super(rankWindowSize);
+        this.inferenceId = inferenceId;
+        this.inferenceText = inferenceText;
+        this.field = field;
+        this.minScore = minScore;
+    }
+
+    public TextSimilarityRankBuilder(StreamInput in) throws IOException {
+        super(in);
+        this.inferenceId = in.readString();
+        this.inferenceText = in.readString();
+        this.field = in.readString();
+        this.minScore = in.readFloat();
+    }
+
+    @Override
+    public String getWriteableName() {
+        return "text_similarity_rank_builder";
+    }
+
+    @Override
+    public TransportVersion getMinimalSupportedVersion() {
+        return TransportVersions.TEXT_SIMILARITY_RERANKER_RETRIEVER;
+    }
+
+    @Override
+    public void doWriteTo(StreamOutput out) throws IOException {
+        out.writeString(inferenceId);
+        out.writeString(inferenceText);
+        out.writeString(field);
+        out.writeFloat(minScore);
+    }
+
+    @Override
+    public void doXContent(XContentBuilder builder, Params params) throws IOException {
+        throw new UnsupportedOperationException("not supported");
+    }
+
+    @Override
+    public boolean isCompoundBuilder() {
+        return false;
+    }
+
+    @Override
+    public Explanation explainHit(Explanation baseExplanation, RankDoc scoreDoc, List<String> queryNames) {
+        return null;
```

Comment:
We could implement something similar to rrf's explain to provide explain info on the match (e..g include info on the inferenceId, initial score etc).

Reply:
Sounds good, I'm working on this.

Reply:
`GET /wikipedia_elser_demo/_explain/17800` gives me a parsing exception: `request does not support [retriever]`. Is there any other place we need to declare the supported fields for the `_explain` API?